sunxspex/sunxspex_fitting/instruments.py

```diff
@@ -1522,11 +1488,58 @@ def _getsrm(self, f_srm):
 
         Returns
         -------
-        A 2d array of the photon and channel bin edges (photon_bins, channel_bins), number of sub-set channels
-        in the energy bin (ngrp), starting index of each sub-set of channels (fchan), number of channels in each
-        sub-set (nchan), 2d array that is the spectral response (srm).
+        SrmFileInfo object
         """
-        return stix_spec._get_srm_file_info(f_srm)
+        ret = stix_spec.load_srm(f_srm, self.srm_type)
+        # trim counts bins to SRM counts bins
+        # (some aren't computed for the SRM)
+        self.clip_energy_edges(ret)
+        return ret
+
+    def clip_energy_edges(self, srm_info: SrmFileInfo):
+        '''STIX SRM bin edges don't always line up with the spectrum edges.
+           Clip them so that matrix multiplication works.
+        '''
+        if srm_info.count_bin_edges.size > self.spec_info.count_bins.size:
```

Comment:
We will very shortly if be moving to a much finer grid in the DRM which will never match the counts bins.